pkg/kubelet/apis/deviceplugin/v1alpha/api.proto

```diff
@@ -24,6 +24,11 @@ service Registration {
 	rpc Register(RegisterRequest) returns (Empty) {}
 }
 
+message DevicePluginOptions {
+        // Indicates if PreStart call is required before each container start
+	bool pre_start_required = 1;
```

Comment:
Just thinking aloud - what happens if we don't have this config? Given plugin communication is localhost, posting an extra RPC might not be too expensive. Thoughts?

Reply:
I think the main reason we want to make this optional is to avoid fail restarted container if device plugin fails.

Reply:
And adding to Jiaying,  all plugins may not need a PreStart on each container start, so giving low latency on restart a preference, this option is disabled by default, means PreStart will be invoked only if asked explicitly.